parsers/nmparser/diagonal_elements.go

```diff
@@ -5,16 +5,42 @@ import (
 	"math"
 )
 
+// lowerDiagonalLengthToDimension takes l which is
+// the number of total elements in the lower diagonal
+//  matrix, and calculates the dimension (aka number of
+// diagonal elements) for the matrix.
+// Additionally, this function can be used to pass
+// a specific index from the array of lower diagonal
+// elements (as l). In that case, it will return one
+// of the following:
+// * If the index is a diagonal element:
+//   * The number of that diagonal element
+//   * true
+// * If the index is not a diagonal element:
+//   * 0
+//   * false
+// For example, there are 6 elements in the lower
+// diagonal of a 3x3 matrix, the 1st, 3rd, and 6th
+// being diagonals. Calling this function on 1-6
+// would return the following:
+// lowerDiagonalLengthToDimension(1) => 1, true
+// lowerDiagonalLengthToDimension(2) => 0, false
+// lowerDiagonalLengthToDimension(3) => 2, true
+// lowerDiagonalLengthToDimension(4) => 0, false
+// lowerDiagonalLengthToDimension(5) => 0, false
+// lowerDiagonalLengthToDimension(6) => 3, true
```

Comment:
this is unrelated to this issue; it's just a final tweak (and some docs) that I was trying to get into #189 but @dpastoor merged it before I could. So quick.